java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java

```diff
@@ -621,8 +622,14 @@ private K disjunctResults(List<K> results) {
                 if (guarded) {
                     ConstrainedTerm result = applySpecRules(term, specRules);
                     if (result != null) {
-                        if (visited.add(result))
+                        if (visited.add(result)) {
                             nextQueue.add(result);
+                        } else {
+                            if (term.equals(result)) {
+                                kem.registerCriticalWarning("Step " + step + ": infinite loop after applying a spec rule.");
+                                proofResults.add(result);
```

Comment:
this is not a bug, this is intended: we only verify properties of terminating executions (partial correctness). We could check that all executions terminate (total correctness), using ranking functions. However, this checks a very particular case, so I'm ok we keeping the warning, but not with adding it to `proofResults`.